GameUI.py

```diff
@@ -62,34 +62,52 @@ def get_instructions(self):
         return user_input
 
     def display_response(self, response):
-        response_mode = response[0]
+        response_mode = response.current_mode
         if response[-1]:
             unique_message = "You already guessed this letter. \n" \
-                             " There are {} more guesses left".format(response[2])
+                             " There are {} more guesses left".format(response.guesses_left)
         elif response_mode == TerminalUI.DISPLAY_DICT['guess_success']:
             unique_message = "GOOD JOB!! \n" \
-                             " Current word" + str(response[1]) \
-                             + " There are {} more guesses left".format(response[2])
+                             " Current word" + str(response.output) \
+                             + " There are {} more guesses left".format(response.guesses_left)
         elif response_mode == TerminalUI.DISPLAY_DICT['guess_failure']:
             self.current_display = self.stages_display.pop(0)
             if not self.current_display:
                 self.current_display = TerminalUI.HANGEDMAN
-            unique_message = "WRONG !! \n Current word" + str(response[1]) + \
-                             " There are {} more guesses left".format(response[2])
+            unique_message = "WRONG !! \n Current word" + str(response.output) + \
+                             " There are {} more guesses left".format(response.guesses_left)
         elif response_mode == TerminalUI.DISPLAY_DICT['hint']:
-            unique_message = "The hint is :", response[1]
+            unique_message = "The hint is :", response.output
+        elif response_mode == TerminalUI.DISPLAY_DICT['solution']:
+            unique_message = "The solution ", response.output
         else:
-            unique_message = "The solution ", response[1]
+            unique_message = "That was an invalid guess. Try again."
 
         for line in self.current_display:
             print(line)
         print(unique_message)
+        return unique_message
 
     def end_UI(self):
         print(TerminalUI.BYE_MESSAGE)
         return
 
 
+class TestUI(GameUI):
```

Comment:
This class derives from `GameUI` but uses fields from `TerminalUI`. It's not clear what this calls means when we have a GUI implementation of the UI.
Also, it should probably be part of the `tests` module, because we use it only for tests and it make no sense inside production code. Same thing for `DummyMethod` and `EmptyMethod`.